server/cmd/mattermost/commands/server.go

```diff
@@ -109,6 +109,10 @@ func runServer(configStore *config.Store, interruptChan chan os.Signal) error {
 
 	notifyReady()
 
+	// Wiping off any signal handlers set before.
+	// This may come from intermediary signal handlers requiring to clean
+	// up resources before server.Start can finish.
+	signal.Reset(syscall.SIGINT, syscall.SIGTERM)
```

Comment:
Is there a chance of missing a signal due to resetting here and "subscribing" again on the following line?

Reply:
There is a chance, yes. But the API does not allow for an atomic change of handlers. You can call `Notify` without calling Reset, to add a new channel handler. But it won't wipe the old handler. I would say this is the best we can do.